File: test-project-data.App/ClientApp/src/Storage/UserRepository.ts
import { User } from "../Models/User";
import { IUserRepository } from "../Interfaces/IUserRepository";
import axios from "axios";

export class UserRepository implements IUserRepository {

    GetUser = async (id: string): Promise<User> => {
        return axios.get<User>(`https://localhost:7254/api/user/${id}`)
            .then(response => {
                if (response.status === 200)
                    return response.data;
                else {
                    console.log("Request failed");
                    return {} as User
                }
            })

    }

    GetUsers = async (): Promise<User[]> => {
        return axios.get<User[]>('https://localhost:7254/api/user')
            .then(response => {
                if (response.status === 200) {
                    console.log("Received:", response.data)
                    return response.data;
                }
                else {
                    console.log("Request failed");
                    return [] as User[]
                }
            })
    }

    CreateUser = async (user: User): Promise<User> => {
        return axios.post<User>('https://localhost:7254/api/user',
            {
                name: user.name,
                vorname: user.vorname,
                e_mail: user.e_mail,
                telefon: user.telefon,
                geburtstag: user.geburtstag
            }
        ).then(response => {
            if (response.status === 201)
                return response.data;
            else {
                throw Error("Request failed");
            }
        })
    }

    UpdateUser = async (user: User): Promise<User> => {
        return axios.put<User>(`https://localhost:7254/api/user/${user.id}`,
            {
                name: user.name,
                vorname: user.vorname,
                e_mail: user.e_mail,
                telefon: user.telefon,
                geburtstag: user.geburtstag
            }
        ).then(response => {
            if (response.status === 204)
                return response.data;
            else {
                throw Error("Request failed");
            }
        })
    }

    DeleteUser = async (id: string): Promise<boolean> => {
        return axios.delete<User>(`https://localhost:7254/api/user/${id}`)
            .then(response => {
                if (response.status === 204)
                    return true;
                else {
                    return false;
                }
            })

    }
}

File: test-project-data.App/Storage/UserRepository.cs
using System.Data;
using test_project_data.App.Models;
using test_project_data.App.Interfaces;
using System;
using System.Collections.Generic;
using Dapper;

namespace test_project_data.App.Storage
{
    public class UserRepository : IUserRepository
    {
        private readonly IDbConnection _db;

        public UserRepository(IDbConnection db)
        {
            _db = db;
        }

        public User GetUser(Guid id) => _db.Get<User>(id);

        public IEnumerable<User> GetUsers()
        {
            return _db.GetList<User>();
        }

        public User CreateUser(User user)
        {
            user.id = _db.Insert<Guid, User>(user);
            return user;
        }

        public void UpdateUser(User user)
        {
            _db.Update(user);
        }

        public void DeleteUser(Guid id)
        {
            var user = _db.Get<User>(id);
            _db.Delete(user);
        }
    }
}